numpy/array_api/_manipulation_functions.py

```diff
@@ -53,13 +53,24 @@ def permute_dims(x: Array, /, axes: Tuple[int, ...]) -> Array:
 
 
 # Note: the optional argument is called 'shape', not 'newshape'
-def reshape(x: Array, /, shape: Tuple[int, ...]) -> Array:
+def reshape(x: Array, 
+            /, 
+            shape: Tuple[int, ...],
+            *,
+            copy: Optional[Bool] = None) -> Array:
     """
     Array API compatible wrapper for :py:func:`np.reshape <numpy.reshape>`.
 
     See its docstring for more information.
     """
-    return Array._new(np.reshape(x._array, shape))
+    if copy is False:
+        raise NotImplementedError("copy=False is not yet implemented")
```

Comment:
I think this can be implemented by calling `reshape` and raising an exception if the result is not a view. Or maybe there's even a more direct way to tell if `reshape` will copy without actually trying it.

Reply:
I guess we need to implement it directly in `np.reshape`. Do others have issues with the above simple workaround here for now though?

Reply:
> I guess we need to implement it directly in np.reshape.

I think that in order to do this correctly, I might have to change the signature of the `PyArray_Newshape(PyArrayObject *self, PyArray_Dims *newdims, NPY_ORDER order)` function in `numpy/core/src/multiarray/shape.c`, but this has been marked as `NUMPY_API` in the comments.

I'll make the change you suggested above, but if I want to go ahead with the change to `PyArray_Newshape`, what do I have to do to get approval for that?

Reply:
@asmeurer Do these changes look OK?